kernel/arch/dreamcast/kernel/cache.s

```diff
@@ -90,6 +97,11 @@ _icache_flush_range:
 ! r5 is count
     .align 2
 _dcache_inval_range:
+    ! Exit early if no blocks to inval
+    mov      #0, r0
+    cmp/eq   r0, r5
+    bt       .dinval_exit
+
     ! Get ending address from count and align start address
     add      r4, r5
     mov.l    align_mask, r0
```

Comment:
```
tst r5,r5
add r4,r5
bt/s .dinval_exit
mov.l align_mask, r0
```
Just one cycle penalty.

Reply:
Actually even better:
```
tst r5,r5
mov.l align_mask, r0
bt .dinval_exit
add r4,r5
```
Then you avoid the stall cycle caused by using r0 right after it's loaded from memory.